addons/pos_sale/models/product_template.py

```diff
@@ -4,6 +4,17 @@
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
+    def get_product_info_pos(self, price, quantity, pos_config_id):
+        res = super().get_product_info_pos(price, quantity, pos_config_id)
+
+        # Optional products
+        res['optional_products'] = [
+            {'name': p.name, 'price': min(p.product_variant_ids.mapped('lst_price'))}
+            for p in self.optional_product_ids.filtered_domain(self._optional_product_pos_domain())
```

Comment:
`_optional_product_pos_domain` is defined in `product.product` and use in `product.template`, shouldn't you adapt that ?